tests/tck/features/optimizer/PushSampleDownRule.feature

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2021 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+Feature: Push Limit down rule
+
+  Background:
+    Given a graph with space named "nba"
+
+  Scenario: push limit down to GetNeighbors
+    When profiling query:
+      """
+      GO 1 STEPS FROM "James Harden" OVER like REVERSELY LIMIT [2]
+      """
+    Then the result should be, in any order:
+      | like._dst         |
+      | "Dejounte Murray" |
+      | "Luka Doncic"     |
+    And the execution plan should be:
+      | id | name         | dependencies | operator info  |
+      | 5  | Project      | 6            |                |
+      | 6  | Limit        | 7            |                |
+      | 7  | GetNeighbors | 0            | {"limit": "2"} |
+      | 0  | Start        |              |                |
+    When profiling query:
+      """
+      GO 2 STEPS FROM "James Harden" OVER like REVERSELY LIMIT [2, 2]
```

Comment:
Is this statement very similar to `GO 2 STEPS FROM "James Harden" OVER like REVERSELY SAMPLE [2, 2] | LIMIT 2`? The only difference is the sampling algorithm, right?

Reply:
Not. Limit clause just perform the final result.

Reply:
And the GO step limit/sample perform on the result of GN

Reply:
I mean `sample[2,2]+limit 2` and `limit [2,2]`, Is there any specific difference in the execution plan between them?